### PR TITLE
Add Subscription#expires_in

### LINKED ISSUE #2775: Pub/Sub ExpirationPolicy?
PR #2764 added ExpirationPolicy to the low level API. Does this need to be exposed in the public API?

### PR DESCRIPTION
**Pull request by blowmage**
Add `expires_in` getter and setter methods that get/set the `ttl` value on ExpirationPolicy.

I decided to expose `expires_in` instead of an ExpirationPolicy because there is only the one field on ExpirationPolicy. If/when ExpirationPolicy adds more fields we can expose this structure while still allowing `expires_in` to directly access that value.

[closes #2775]

### CODE REVIEW

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb [hunk 1]
```diff
@@ -256,6 +256,47 @@ def labels= new_labels
           @resource_name = nil
         end
 
+        ##
+        # The duration (in seconds) for when a subscription expires after the
+        # subscription goes inactive. A subscription is considered active as
+        # long as any connected subscriber is successfully consuming messages
+        # from the subscription or is issuing operations on the subscription.
+        #
+        # If `expires_in` is not set, a *default* value of of 31 days will be
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: link to `{#expires_in=}` here?

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb [hunk 2]
```diff
@@ -256,6 +256,47 @@ def labels= new_labels
...
+          ensure_grpc!
+
+          return nil if @grpc.expiration_policy.nil?
+
+          Convert.duration_to_number @grpc.expiration_policy.ttl
+        end
+
+        ##
+        # Sets the duration (in seconds) for when a subscription expires after
+        # the subscription goes inactive.
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: Add "See also #{expires_in}" here?

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb [hunk 3]
```diff
@@ -256,6 +256,47 @@ def labels= new_labels
...
+        #
+        # @param [Numeric, nil] ttl The expiration duration, or `nil` to unset.
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: what is the unit? days? seconds? millis?

Comment (quartzmo) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: Ah, I see now `seconds` in the description, sorry. But wouldn't hurt to repeat it.

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb (hunk 1 above)
Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: Added.

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb (hunk 2 above)
Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: Added.

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb (hunk 3 above)
Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: Added.

**Pull request merged**